modules/arm_plugin/src/arm_converter/arm_converter_one_hot.cpp

```diff
@@ -2,88 +2,111 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include <details/ie_exception.hpp>
-
 #include "arm_converter/arm_converter.hpp"
 #include <ngraph/runtime/reference/one_hot.hpp>
 
-namespace ArmPlugin {
-template <typename Indices, typename OutputType>
-void wrap_one_hot(const Indices* arg,
+template <typename InputType, typename OutputType>
+void wrap_one_hot(const InputType* indices,
+                  const ngraph::Shape& indices_shape,
                   OutputType* out,
-                  const ngraph::Shape& in_shape,
-                  const ngraph::Shape& out_shape,
-                  size_t one_hot_axis,
-                  const OutputType* on_values,
-                  const OutputType* off_values) {
-    ngraph::runtime::reference::one_hot(arg,
-                                        out,
-                                        in_shape,
-                                        out_shape,
-                                        one_hot_axis,
-                                        on_values[0],
-                                        off_values[0]);
+                  const size_t out_elem_size,
+                  const std::int64_t depth,
```

Comment:
```suggestion
                  const InputType* depth,
```

and extract the depth value below. With these changes, we will be able to work with inconstant depth.

Reply:
Thank you! Your suggestion was incorporated in the latest commit.